mlflow/tracking/fluent.py

```diff
@@ -2443,5 +2473,7 @@ def setup_autologging(module):
         setup_autologging(pyspark_module)
         setup_autologging(pyspark_ml_module)
     else:
-        register_post_import_hook(setup_autologging, "pyspark", overwrite=True)
-        register_post_import_hook(setup_autologging, "pyspark.ml", overwrite=True)
+        if "spark" in target_library_and_module:
```

Comment:
implement in this way due to the slight difference between library name ("pyspark") and flavor name ("spark")